Sources/Yams/Tag.swift

```diff
@@ -0,0 +1,120 @@
+         _ resolver: Resolver,
+         _ constructor: Constructor) {
+        self.resolver = resolver
+        self.constructor = constructor
+        if let string = string, !string.isEmpty && string != "!" {
+            name = Name(rawValue: string)
+        } else {
+            name = nil
+        }
+    }
+
+    init(_ name: Name,
+         _ resolver: Resolver,
+         _ constructor: Constructor) {
+        self.resolver = resolver
+        self.constructor = constructor
+        self.name = name
+    }
+
+    func resolved(with node: Node) -> Tag {
+        if name == nil {
+            name = resolver.resolveTag(of: node)
+        }
+        return self
+    }
+
+    static var implicit: Tag {
+        return Tag(nil, .default, .default)
+    }
+
+    // fileprivate
+    fileprivate let resolver: Resolver
+}
+
+extension Tag: Hashable {
+    public var hashValue: Int {
+        return name?.hashValue ?? 1
+    }
+
+    public static func == (lhs: Tag, rhs: Tag) -> Bool {
+        switch (lhs.name, rhs.name) {
+        case let (lhs?, rhs?): return lhs == rhs
+        case (.none, _): fallthrough
+        case (_, .none): fatalError("Never happen this!")
+        default: return false
+        }
+    }
+}
+
+extension Tag.Name: ExpressibleByStringLiteral {
+    public init(stringLiteral value: String) {
+        self.rawValue = value
+    }
+
+    public init(unicodeScalarLiteral value: String) {
+        self.rawValue = value
+    }
+
+    public init(extendedGraphemeClusterLiteral value: String) {
+        self.rawValue = value
+    }
+}
+
+extension Tag.Name: Hashable {
+    public var hashValue: Int {
+        return rawValue.hashValue
+    }
+
+    public static func == (lhs: Tag.Name, rhs: Tag.Name) -> Bool {
+        return lhs.rawValue == rhs.rawValue
+    }
+}
+
+// http://www.yaml.org/spec/1.2/spec.html#Schema
+extension Tag.Name {
+    // Failsafe Schema
+    public static let str: Tag.Name = "tag:yaml.org,2002:str"
+    public static let seq: Tag.Name  = "tag:yaml.org,2002:seq"
+    public static let map: Tag.Name  = "tag:yaml.org,2002:map"
+    // JSON Schema
+    public static let bool: Tag.Name  = "tag:yaml.org,2002:bool"
+    public static let float: Tag.Name  =  "tag:yaml.org,2002:float"
+    public static let null: Tag.Name  = "tag:yaml.org,2002:null"
+    public static let int: Tag.Name  = "tag:yaml.org,2002:int"
+    // http://yaml.org/type/index.html
+    public static let binary: Tag.Name  = "tag:yaml.org,2002:binary"
+    public static let merge: Tag.Name  = "tag:yaml.org,2002:merge"
+    public static let omap: Tag.Name  = "tag:yaml.org,2002:omap"
+    public static let pairs: Tag.Name  = "tag:yaml.org,2002:pairs"
+    public static let set: Tag.Name  = "tag:yaml.org,2002:set"
+    public static let timestamp: Tag.Name  = "tag:yaml.org,2002:timestamp"
+    public static let value: Tag.Name  = "tag:yaml.org,2002:value"
+//    public static let yaml: Tag.Name  = "tag:yaml.org,2002:yaml"
```

Comment:
I could not decide how we handle this.
PyYAML is not implemented to handling "tag:yaml.org,2002:yaml".
When I read [the specification](http://yaml.org/type/yaml.html) again, it seems that it can not be implemented.
I'll change that comment to declaration and add comment describing we don't support that.